src/client/index.ts

```diff
@@ -0,0 +1,55 @@
+import { WalletProviderResponseError } from "../utils/errors";
+import {
+  ProblemDetail,
+  createApiClient as createWalletProviderApiClient,
+} from "./generated/wallet-provider";
+import { ApiClient as WalletProviderApiClient } from "./generated/wallet-provider";
+
+export type WalletProviderClient = WalletProviderApiClient;
+
+const validateResponse = async (response: Response) => {
+  if (!response.ok) {
+    var problemDetail: ProblemDetail = {};
+    try {
+      problemDetail = ProblemDetail.parse(await response.json());
+    } catch (_) {
+      problemDetail = {
+        title: "Invalid response from Wallet Provider",
+        detail: "Unable to parse response to ProblemDetail",
+      };
+    }
+
+    throw new WalletProviderResponseError(
+      problemDetail.title
+        ? problemDetail.title
+        : "Invalid response from Wallet Provider",
+      problemDetail.type,
+      problemDetail.detail
+    );
+  }
+  return response;
+};
+
+export function getWalletProviderClient(context: {
```

Comment:
I would use the same method for defining functions (all through function declarations or through function expressions)

Reply:
Right, thanks